python_modules/dagster/dagster/_config/pythonic_config/typing_utils.py

```diff
@@ -27,6 +28,18 @@
     from dagster._config.pythonic_config import PartialResource
 
 
+def _materialize_annotations_for_pydantic(namespaces: dict[str, Any]) -> dict[str, Any]:
+    """Materialize lazy annotations in Python 3.14+ (PEP 649)."""
+    annotations = namespaces.get("__annotations__", {})
+
+    # if annotations are empty, try calling __annotate_func__ (PEP 649)
+    if sys.version_info >= (3, 14) and not annotations and "__annotate_func__" in namespaces:
+        annotations = namespaces["__annotate_func__"](1)  # format=1 for VALUE
+        namespaces["__annotations__"] = annotations
+
+    return annotations
```

Comment:
I dont see `__annotate_func__` in the pep documentation 

should we use https://docs.python.org/3/library/annotationlib.html instead? 

something along the lines of 
```
annotationlib.get_annotations(
  namespaces,
  format=annotationlib.Format.VALUE,
)
```

Reply:
I'm confused by this comment, was there supposed to be a code change? 

My top concern is the use of `__annotate_func__`. I cant find any reference for it in pep / python docs and top result when searching is this issue about how it should not be used https://github.com/astral-sh/ruff/issues/17859

Secondary is using annotationlib instead of manually processing the annotate fn, since it seems like it exists to handle this. What issues were you running in to exactly?

Reply:
lol sorry I thought I pushed up the changes but I did not -- i do end up using annotationlib directly, just needed to call it in a particular way